src/Foundatio.Redis/Storage/RedisFileStorage.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Foundatio.Extensions;
+using Foundatio.Serializer;
+using Foundatio.Utility;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using StackExchange.Redis;
+
+namespace Foundatio.Storage {
+    public class RedisFileStorage : IFileStorage {
+        private readonly RedisFileStorageOptions _options;
+        private readonly ILogger _logger;
+        private readonly string _fileSpecContainer;
+
+        public RedisFileStorage(RedisFileStorageOptions options) {
+            if (options.ConnectionMultiplexer == null)
+                throw new ArgumentException("ConnectionMultiplexer is required.");
+            options.Serializer = options.Serializer ?? DefaultSerializer.Instance;
+            _logger = options.LoggerFactory?.CreateLogger(typeof(RedisFileStorage)) ?? NullLogger.Instance;
+            options.ConnectionMultiplexer.ConnectionRestored += ConnectionMultiplexerOnConnectionRestored;
+            _fileSpecContainer = $"{options.ContainerName}-filespecs";
```

Comment:
Is there any reason this doesn't just used the passed in container name? What if the container name is null?

Reply:
This is where additional filespec info is being stored. The files are being stored in the containername that was specified. But we probably should have a null check here for ContainerName.